### PR TITLE
MVar implementation

### PR DESCRIPTION
**Pull request by ani003**
This PR adds an implementation of MVar in wasm based on the [OCaml example](https://github.com/ocaml-multicore/effects-examples/blob/master/mvar/MVar.ml).

### CODE REVIEW

File: test/core/cont.wast [hunk 1]
```diff
@@ -538,3 +538,476 @@
...
+  (event $resume (export "resume") (param (ref null $cont)))
+
+  ;; Table as simple queue (keeping it simple, no ring buffer)
+  (table $curr_proc 0 (ref null $cont))
+  (table $queue 0 (ref null $cont))
+
+  (global $qdelta i32 (i32.const 10))
+  (global $qback (mut i32) (i32.const 0))
+  (global $qfront (mut i32) (i32.const 0))
+  (global $curr_status (mut i32) (i32.const 0))
```
Comment (dhil) [inline test/core/cont.wast]: It would be good to add a comment detailing the purpose of each of these variables.

File: test/core/cont.wast [hunk 2]
```diff
@@ -538,3 +538,476 @@
...
+
+(module $producer_queue
+  (type $proc (func))
+  (type $cont (cont $proc))
+
+  ;; Table as simple queue (keeping it simple, no ring buffer)
+  (table $queue 0 (ref null $cont))
+  (global $qdelta i32 (i32.const 10))
+  (global $qback (mut i32) (i32.const 0))
+  (global $qfront (mut i32) (i32.const 0))
```
Comment (dhil) [inline test/core/cont.wast]: A comment to detail the purpose of these variables would be nice.

File: test/core/cont.wast [hunk 3]
```diff
@@ -538,3 +538,476 @@
...
+
+(module $consumer_queue
+  (type $proc (func))
+  (type $cont (cont $proc))
+
+  ;; Table as simple queue (keeping it simple, no ring buffer)
+  (table $queue 0 (ref null $cont))
+  (global $qdelta i32 (i32.const 10))
+  (global $qback (mut i32) (i32.const 0))
+  (global $qfront (mut i32) (i32.const 0))
```
Comment (dhil) [inline test/core/cont.wast]: Comment here too, please.

**Review by dhil — COMMENTED** [review]
LGTM.

**Comment by ani003** [comment]
@dhil I have added a few comments. Please check if that is sufficient

**Comment by dhil** [comment]
> @dhil I have added a few comments. Please check if that is sufficient

Looks good. I am happy to merge this if you mark the PR as non-draft.

**Comment by ani003** [comment]
@dhil extremely sorry for the delay. I was a bit preoccupied and I forgot about it until now.

**Comment by dhil** [comment]
No need to apologise. It is all good. We all got busy schedules. :-) I will merge it now. Thanks!

**Comment by slindley** [comment]
Oops... I should really have looked at this earlier. This PR shouldn't have been merged because it consists of a big chunk of code in `test/core/cont.wast`. This is the wrong place for a larger example like this. It should really be in `proposals/continuations/examples` where there are many other similar examples.

**Pull request merged**